.github/workflows/ci.yml

```diff
@@ -9,40 +9,64 @@ on:
 jobs:
   build-test:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Install Nix
-        uses: DeterminateSystems/nix-installer-action@v16
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: ./fractal-engine/go.mod
+          cache: true
 
-      - name: Enable Nix cache
-        uses: DeterminateSystems/magic-nix-cache-action@v8
+      - name: Go mod download
+        working-directory: ./fractal-engine
+        run: go mod download
 
-      - name: Run tests
+      - name: Run Go tests
+        working-directory: ./fractal-engine
         env:
-          GO_TEST_EXTRA_FLAGS: "-race"
-        run: nix run .#test
-
-      - name: Build binary
-        run: nix build .#fractalengine
-
-      - name: Prepare artifact
-        id: prepare
-        run: |
-          set -euo pipefail
-          SYS=$(nix eval --raw --impure --expr 'builtins.currentSystem')
-          echo "sys=$SYS" >> $GITHUB_OUTPUT
-          cp -p result/bin/fractal-engine "fractal-engine-$SYS"
+          CGO_ENABLED: 1
+        run: go test ./pkg/... -race -coverprofile=coverage.txt -covermode=atomic
 
       - name: Upload coverage
         uses: actions/upload-artifact@v4
+        if: always()
         with:
           name: coverage
-          path: coverage.txt
+          path: fractal-engine/coverage.txt
 
-      - name: Upload binary
```

Comment:
We don't want to remove this - we still want to publish binaries against releases, alongside OCI containers